controllers/UsersController.js

```diff
@@ -1,92 +1,92 @@
-import { body, validationResult } from 'express-validator/check';
-import { sanitizeBody } from 'express-validator/filter';
-// import debug from 'debug';
+import Model from '../models/Model';
+import { InternalServerError } from '../utils/errorHandlers';
+// import { dev_logger } from '../utils/loggers';
 
-import users from '../utils/sample.users';
-
-// const logger = message => (debug('dev')(message));
+const users_model = new Model('users');
 
 const UsersController = {
 
-    dashboard: (req, res) => {
-        res.render( 'dashboard', { title: 'Dashboard' });
+    get_user: (req, res) => {
+        const { id } = req.params;
+        (async () => {
+            try {
+                const { rows } = await users_model.select(
+                    `id, email, password, firstname,
+                    lastname, phone, status, address`,
+                    `id=${id}`
+                );
+                if (rows.length === 0) {
+                    // user was not found
+                    return res.status(404).json({
+                        error: `User with id ${id} not found`
+                    });
+                }
+                return res.status(200).json({ data: rows[0] });
+            }
+            catch (e) {
+                return InternalServerError(req, res, e);
+            }
+        })();
     },
-    
+
     verify_user: (req, res) => {
-        const { email } = req.params;
-        const user = users.find(user => (user.email === email));
-        
-        if (user === undefined) {
-            res.status(404).json({ error: 'User not found' });
-        }
-        else {
-            user.status = 'verified';
-            res.status(200).json({ data: user });
-        }
+        const { id } = req.params;
+        (async () => {
+            try {
+                await users_model.update(
+                    'status=\'verified\'',
+                    `id=${id}`
+                );
+                UsersController.get_user(req, res);
+            }
+            catch (e) { return InternalServerError(req, res, e); }
+        })();
     },
-    
+
     get_users: (req, res) => {
         const { status } = req.query;
-        
-        if (status) {
-            const data = users.filter(user => (user.status === status));
-            res.status(200).json({ data });
-        }
-        else {
-            const data = [];
-            users.map(user => data.push(user));
-            res.status(200).json({ data });
-        }
+        (async () => {
```

Comment:
Make the outer function async you won't have a need for this IIFE

Reply:
Done